### DISCUSSION TITLE
[RFC] Move cards to "Layout" category.

### DISCUSSION THREAD

**Discussion by aayushpi** [twilio-labs/paste#477] [Review request]
Case:

- Cards are not atomic components i.e. you can't use a card by itself.
- Cards are used exclusively as containers
- Cards define structure just like Grid does.
- Card is entirely a compositional tool.

Counterpoint:

- Cards will be our only visible layout opinion

Counter-to-Counter point:

- Visibility is not a defining component criteria since for many users, `Screen Reader Only` is not visible. Neither is truncate

**Comment by SiTaggart** [comment]
I am conflicted on this one and I think it's purely gut and nothing concrete.

Like you've kind of touched on, layout components typically for me come with no visuals, they're just utility. Where as a Card does come with design opinion baked in, so I've never considered them "layout".

But it's not like a Card is like any other normal component, it's pretty useless on it's own and is used purely for creating structure, very similar to layout components.

So, I'm torn a little between habit and readjusting my view of cards. I could go either way.

**Comment by serifluous** [comment]
Some initial thoughts/discussion points (though I want to think on this a little longer):

- If we do put it in layout, I think it'd also make sense to add Separator to layout.
- Some people closely associate Card and Accordions, and may want to look for them in the same place. When @pguptaDS and Porlin did their Paste library research, they saw people tended to group those two components together in a "Containers" category.
- One thing that may get a little tricky is if we ever end up defining more of the structure and content of a Card (e.g., adding a footer, giving it a default heading), then does it still make sense as a Layout component? Though we could always move it again.

**Reply by SiTaggart** [reply]
I'd be interested to see some research into what makes a site easier to navigate:

- components placed into distinct categories in the IA
- a single flat list of all components and a really good search

I've always wondered if categorizing components is just a fruitless task as so many people will think about components and categories in many different ways. Will you ever be able to create a category system that makes sense to everyone, or should you give up and make a giant, flat searchable list?

**Reply by corinne-nagel** [reply]
I really like this question, this is definitely something we should test when the time comes.

**Reply by SiTaggart** [reply]
Timely related https://github.com/twilio-labs/paste/discussions/484

**Comment by TheSisb** [comment]
I kind of see Card as a visual container, and not necessarily layout. It doesn't position or size its child elements like the other layout components.  It only provides a styling for elements that are positioned using other layout components.

That said, if the definition of layout is positioning child elements, then are Modal and Tabs layout?

**Comment by corinne-nagel** [comment]
Hmmm I think of cards as a layout. But definitely more of a foundational element...I think it's a base that you can add components to and position/size relatively within. But yeah, I would think of Modal and tabs as a layout as well. Anything that is a composition of sorts I would consider that.

**Comment by richbachman** [comment]
I'm on the Card not being a layout component band wagon. A card is a thing that can be placed inside various layouts. For example, a Card can span multiple columns or a single column of a layout.

**Comment by corinne-nagel** [comment]
Circling back here, what do you think the consensus should be here @aayushpi ? Seems like it could go both ways...

**Comment by serifluous** [comment]
Updated website nav based on card-sorting research 👍👍
